### PR TITLE
Add note about ambient dts files to App types

### LINKED ISSUE #3853: Using a import type statement in app.d.ts breaks
### Describe the bug

Currently when I use a import type statement in app.d.ts (where my app types are) it will no longer work. I know it's just this statement as when I use an alternative like `import('magic-sdk').MagicUserMetadata` in the typedef it works fine

### Reproduction

In this example I am using MagicUserMetadata, when I add the import the user and test fields no longer work

```ts
import type { MagicUserMetadata } from 'magic-sdk';

/// <reference types="@sveltejs/kit" />

// See https://kit.svelte.dev/docs#typescript
// for information about these interfaces
declare namespace App {
    interface Locals {
        test: boolean;
        user: MagicUserMetadata;
    }

    interface Platform {}

    interface Session {}

    interface Stuff {}
}
```

### Logs

_No response_

### System Info

```shell
System:
    OS: Linux 5.16 undefined
    CPU: (12) x64 AMD Ryzen 5 3600 6-Core Processor
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.9.1 - ~/.nvm/versions/node/v16.9.1/bin/node
    Yarn: 1.22.17 - ~/.nvm/versions/node/v16.9.1/bin/yarn
    npm: 7.24.2 - ~/.nvm/versions/node/v16.9.1/bin/npm
  Browsers:
    Brave Browser: 98.1.35.100
    Chromium: 98.0.4758.80
    Firefox: 96.0.3
  npmPackages:
    @sveltejs/adapter-vercel: next => 1.0.0-next.43 
    @sveltejs/kit: next => 1.0.0-next.266 
    svelte: ^3.46.4 => 3.46.4
```

### Severity

serious, but I can work around it

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by ghostdevv**
### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [ ] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpx changeset` and following the prompts. All changesets should be `patch` until SvelteKit 1.0

Adds a note about how you can't use `import` statements in the app declaration file, closes #3853

### CODE REVIEW

**Comment by ghostdevv** [comment]
Actually this doesn't need a changeset right?

**Comment by benmccann** [comment]
correct, it'd be better to remove the changeset for this one since there's no user-visible change

**Comment by Rich-Harris** [comment]
Weirdly, this isn't affecting the relevant part of the docs: https://kit-svelte-n0jl1ronp-svelte.vercel.app/docs/types#app

I don't know if that's some quirk of the process that extracts the types, or some turborepo configuration that is causing that step to be skipped

**Comment by ghostdevv** [comment]
Have removed the changeset - no idea how to use turborepo xD

File: packages/kit/types/ambient.d.ts [hunk 1]
```diff
@@ -14,6 +14,23 @@
  * 	interface Stuff {}
  * }
  * ```
+ * 
+ * Note: Since it's an ambient dts file, you can't use `import` statements. Instead you should use the import() function, for example:
```
Comment (benmccann) [inline packages/kit/types/ambient.d.ts:18]: I think if we want to add this note, it'd make more sense to add it at the end. And you could change the colon in the last sentence to a period then

Comment (ghostdevv) [reply packages/kit/types/ambient.d.ts:20]: Sorry it took a week! Have done this @benmccann

**Comment by ghostdevv** [comment]
No idea why these tests are failing

**Review by dummdidumm — APPROVED** [review]

**Comment by Rich-Harris** [comment]
Gah, the docs still aren't updating — Turbo is just replaying caches, even though they should be invalid. Will investigate tomorrow

**Comment by Rich-Harris** [comment]
thank you!

**Comment by ghostdevv** [comment]
Glad you got it sorted! xD np

**Pull request merged**